modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalRequest.java

```diff
@@ -64,16 +69,53 @@ public void setRankEvalSpec(RankEvalSpec task) {
         this.rankingEvaluationSpec = task;
     }
 
+    /**
+     * Sets the indices the search will be executed on.
+     */
+    public RankEvalRequest setIndices(String... indices) {
+        Objects.requireNonNull(indices, "indices must not be null");
+        for (String index : indices) {
+            Objects.requireNonNull(index, "index must not be null");
+        }
+        this.indices = indices;
+        return this;
+    }
+
+    /**
+     * @return the indices for this request
+     */
+    public String[] getIndices() {
+        return indices;
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         rankingEvaluationSpec = new RankEvalSpec(in);
-
+        if (in.getVersion().onOrAfter(Version.V_6_3_0)) {
+            indices = in.readStringArray();
+        } else {
+            // older 6.2 format
+            int indicesSize = in.readInt();
+            String[] indices = new String[indicesSize];
+            for (int i = 0; i < indicesSize; i++) {
+                indices[i] = in.readString();
+            }
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         rankingEvaluationSpec.writeTo(out);
+        if (out.getVersion().onOrAfter(Version.V_6_3_0)) {
```

Comment:
on the version conditionals, you may need to first use 7_0_0_alpha1 on master before you backport to 6.x, then you can update the version on master too.

Reply:
Given that 6.3 isn't release yet, bwc tests should only run vs. 6.2 if this is merged shortly, or am I missung sth?

Reply:
don't we run bwc tests between 6.x and previous versions?

Reply:
np, I just thought about it when you mentioned it.